generate/src/lib.rs

```diff
@@ -60,15 +65,46 @@ impl Builder {
     /// needed if the crate was renamed in the Cargo manifest.
     pub fn with_runtime_crate_name<S>(mut self, name: S) -> Self
     where
-        S: AsRef<str>,
+        S: Into<String>,
     {
-        self.args.runtime_crate_name = name.as_ref().to_owned();
+        self.args.runtime_crate_name = name.into();
+        self
+    }
+
+    /// Manually adds specifies the deployed address of a contract for a given
+    /// network. Note that manually specified deployments take precedence over
+    /// deployments in the Truffle artifact (in the `networks` property of the
+    /// artifact).
+    ///
+    /// This is useful for integration test scenarios where the address of a
+    /// contract on the test node is deterministic (for example using
+    /// `ganache-cli -d`) but the contract address is not part of the Truffle
+    /// artifact; or to override a deployment included in a Truffle artifact.
+    pub fn add_deployment(mut self, network_id: u32, address: Address) -> Self {
+        self.args.deployments.insert(network_id, address);
         self
     }
 
+    /// Manually adds specifies the deployed address as a string of a contract
+    /// for a given network. See `Builder::add_deployment` for more information.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the specified address string is invalid. See
+    /// `parse_address` for more information on the address string format.
+    pub fn add_deployment_str<S>(self, network_id: u32, address: S) -> Self
```

Comment:
Are we using this?

Reply:
No, but it makes it easier to use when using `ethcontract_generate` in `build.rs`. I write this with `dex-services` in mind, to avoid having to write the `parse_address` code there.

The idea is that `dex-services` will call this with the deterministic contract addresses so that there is no more requirement to run `truffle migrate` before building.